tools/browser_compat_data_update.py

```diff
@@ -0,0 +1,113 @@
+                unknown_keys = set(support.keys()) - {
+                    "version_added",
+                    "notes",
+                    "alternative_name",
+                }
+                if unknown_keys:
+                    # Assume unsupported if feature has been removed after it was added, requires flags or something like that.
+                    continue
+                out_value[browser] = parse_version(version_added.lstrip("≤"), browser)
+            # Don't bother adding JS features that have ~always been supported, we're not going to feature-test those.
+            if not all(
+                min_interesting_version >= out_value.get(browser, maxsize)
+                for browser, min_interesting_version in min_interesting_versions.items()
+            ):
+                out["#"] = out_value
+    return out
+
+
+out = {}
+
+# Pull JavaScript builtins support from MDN's Browser Compat Data.
+data = requests.get("https://unpkg.com/@mdn/browser-compat-data/data.json").json()
+# Fill minimum browser versions based on year.
+min_interesting_versions = {
+    browser: min(
+        parse_version(version, browser)
+        for version, release in data["browsers"][browser]["releases"].items()
+        if "release_date" in release and int(release["release_date"][:4]) >= min_interesting_year
+    )
+    for browser in interesting_browsers
+}
+print("Minimum interesting versions: ", min_interesting_versions)
+out["js"] = traverse_bcd(data["javascript"]["builtins"])
+# There are a lot of browser APIs and very few we care about, so only pull them to keep JSON small.
+for api in ["AudioWorklet"]:
+    # Just add them to the same JS namespace.
+    out["js"][api] = traverse_bcd(data["api"][api])
+# Manually add 'performance' because browser-compat-data currently doesn't match https://caniuse.com/high-resolution-time.
+# It shows when API is available, but not when it returns actually high-resolution time.
+out["js"]["performance"] = {
+    "#": {"chrome": 24, "firefox": 15, "nodejs": 160000, "safari": 80000}
+}
+
+# BCD data is not as complete for Wasm features as webassembly.org, at least for now.
+data = requests.get("https://webassembly.org/features.json").json()
+out_wasm = out["wasm"] = {}
+for feature in data["features"]:
+    out_wasm[feature] = {"#": {}}
+for browser in ["Chrome", "Firefox", "Safari", "Node.js"]:
+    normalized_browser = browser.lower().replace(".", "")
+    for feature, support in data["browsers"][browser]["features"].items():
+        if type(support) is list:
+            # If support is a list, first item is the version and 2nd is a text note.
+            support = support[0]
+        if type(support) is not str or support == "flag":
+            continue
+        out_wasm[feature]["#"][normalized_browser] = parse_version(
+            support, normalized_browser
+        )
+
+with open(Path(__file__).parent / "browser_compat_data.json", "w") as f:
+    json.dump(out, f, indent="\t")
```

Comment:
We use 2-space indentation in python scripts in this repo

Reply:
How about avoiding tabs in the json and using 2 or 4 spaces instead?

Reply:
As for JSON, if would increase size somewhat more and I figured it might be not worth it.

Reply:
If you limit the json to just the features we care about that should no longer matter.

Reply:
I'm really not a fan of that as it requires to remember to carefully add features to a separate list to avoid errors, and because it limits usefulness in 3rd-party libs...

Reply:
Given the existing size of the repo, 100KB for checkout really seems like not a lot, and once checked in, it changes very rarely for existing features so much lower maintenance overhead.

Reply:
Its not just about the overall size, but also the ability to maintain and review.  For example, imagine when we update this from time to time and all kind of things will change.  I'd rather keep to limits and precise.

Third party library authors can request more over time, and if that happen more than N times a year we can reconsider.

The other nice thing about enumerating the full list of features we need is that the update process would fail right away if any feature is missing/renamed.

Reply:
> For example, imagine when we update this from time to time and all kind of things will change.

Ideally you shouldn't review the auto-generated file. I mean, we could as well generate it as part of `bootstrap`, the only reason it's checked in is to avoid extra network requests.

> The other nice thing about enumerating the full list of features we need is that the update process would fail right away if any feature is missing/renamed.

That should never happen. A lot of tools and websites - caniuse, Babel, etc - depend on this database.